packages/compiler/src/render3/view/template.ts

```diff
@@ -727,10 +713,12 @@ export class TemplateDefinitionBuilder implements t.Visitor<void>, LocalResolver
         const hasValue = value instanceof LiteralPrimitive ? !!value.value : true;
         this.allocateBindingSlots(value);
         const bindingName = prepareSyntheticPropertyName(input.name);
-        this.updateInstruction(elementIndex, input.sourceSpan, R3.elementProperty, () => {
+
+        this.updateInstruction(elementIndex, input.sourceSpan, R3.property, () => {
           return [
-            o.literal(elementIndex), o.literal(bindingName),
-            (hasValue ? this.convertPropertyBinding(implicit, value) : emptyValueBindInstruction)
+            o.literal(bindingName),
+            (hasValue ? this.convertPropertyBinding(implicit, value, true) :
```

Comment:
Nit: for readability. Same for boolean args below

```suggestion
            (hasValue ? this.convertPropertyBinding(implicit, value, /* skipBindFn */ true) :
```